src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1948,7 +1949,13 @@ private FeedbackSessionResultsBundle getFeedbackSessionResultsForUserWithParams(
                         }
                         isVisibleResponse = false;
                     }
-
+                }
+                
+                if (question.questionType == FeedbackQuestionType.CONTRIB) {
```

Comment:
Instead of doing that check in the questionDetails, maybe you can modify the CourseRoster passed into <code>FeedbackSessionResultsBundle</code> or the method that constructs <code>rosterSectionTeamNameTable</code>. It seems that the logic for obtaining the course roster in CoursesLogic also doesn't consider instructor permissions, we should fix that in another issue?